mobile/ios/ios-core.ts

```diff
@@ -980,44 +984,100 @@ export class PlistService implements Mobile.IiOSDeviceSocket {
 	}
 }
 
-export class GDBServer implements Mobile.IGDBServer {
-	private socket: Mobile.IiOSDeviceSocket  = null;
+function getCharacterCodePoint(ch: string) {
+	assert.equal(ch.length, 1);
 
-	constructor(private service: number,
-		private $injector: IInjector) {
+	var code = ch.charCodeAt(0);
+	assert.ok(!(0xD800 <= code && code <= 0xffff));
 
-		if(hostInfo.isWindows()) {
-			this.socket = this.$injector.resolve(WinSocket, {service: this.service, format: 0});
-		} else if(hostInfo.isDarwin()) {
-			this.socket = this.$injector.resolve(PosixSocket, {service: this.service, format: CoreTypes.kCFPropertyListXMLFormat_v1_0});
+	return code;
+}
+
+util.inherits(GDBStandardOutputAdapter, stream.Transform);
+function GDBStandardOutputAdapter(opts?: stream.TransformOptions) {
+	stream.Transform.call(this, opts);
+	this.utf8StringDecoder = new string_decoder.StringDecoder("utf8");
+}
+GDBStandardOutputAdapter.prototype._transform = function(packet: Buffer, encoding: string, done: Function): void {
+	var result = "";
+
+	for (var i = 0; i < packet.length; i++) {
+		if (packet[i] === getCharacterCodePoint("$")) {
```

Comment:
This comparison happens on plenty of places. What is your opinion about a helper function
``` codePointCompare(byte: number, char: string) => byte === getCharacterCodePoint(char)``` ?